src/core/tree_types.ml

```diff
@@ -61,12 +61,13 @@ type tree_cond =
 type 'rhs tree =
   | Fail
   (** Empty decision tree, used when there are no rewriting rules. *)
-  | Leaf of (int * (int * int array)) list * 'rhs
-  (** The value [Leaf(m, rhs)] stores the RHS [rhs] of the rewriting rule that
+  | Leaf of (int * (int * int array)) list * 'rhs * int
+  (** Triple [Leaf(m, rhs, x)] stores the RHS [rhs] of the rewriting rule that
```

Comment:
As mentioned previously, more details on the meaning of the elements of [m] would be very welcome here.

Reply:
Tell me if the documentation isn't clear enough.